src/etc/rc.initial

```diff
@@ -151,18 +148,21 @@ case ${opmode} in
 	/etc/rc.php-fpm_restart
 	;;
 100)
-	if grep "$WORD" "$CONFIG"; then
-		links "https://localhost"
-	else
-		links "http://localhost"
+	protocol=$(/usr/local/sbin/read_xml_tag.sh string system/webgui/protocol)
+	port=$(/usr/local/sbin/read_xml_tag.sh string system/webgui/port)
+	if [ -z "$port" ]; then
+		case $protocol in
+			https) port=443;;
+			*)     port=80;;
+		esac
 	fi
+	links "${protocol}://localhost:${port}"
```

Comment:
`protocol` could be empty, so it needs an extra test to drop a default value in there if it is (`http`).

Reply:
Didn't know protocol could be empty. Ok I will rework to account for that.